src/FileSystem/FilesFinder.php

```diff
@@ -66,6 +66,8 @@ private function findInDirectories(array $directories, array $suffixes, bool $so
             ->files()
             // skip empty files
             ->size('> 0')
+            // changes in PHP files with short echo tag will mostly create invalid code, tags are required
+            ->notContains('<?=')
```

Comment:
it probably better to make it configurable, disallow by default

Reply:
This is fix to current behavior. Rector now changes `<?=` code:

```diff
+<?php
+
+declare(strict_types=1);
 <?= $something ?>
```

Which will cause code to crash.

Reply:
File paths collection is not only on `findInDirectories()` method, but also `filterFiles()`, see usage on `findInDirectoriesAndFiles()` method on line 29, as direct file path will still not checked

Reply:
> Also, <?= is probably just a string, probably exclude with "not start with", also wondering of possible bc break on downgrade.

Makes sense :+1:

Reply:
Ref https://github.com/rectorphp/rector/issues/8425, issue that triggered this.

Reply:
Skipping the files by default is too much default magic IMO.

could rector at least emit a warning and report which file has which problem?

Reply:
@staabm Agreed - https://github.com/rectorphp/rector-src/pull/6068#issuecomment-2197162261